lib/spack/spack/bootstrap/core.py

```diff
@@ -559,6 +559,21 @@ def ensure_patchelf_in_path_or_raise() -> spack.util.executable.Executable:
         )
 
 
+def ensure_winsdk_or_raise() -> None:
+    """Ensure the Windows SDK + WGL are available on system"""
+    externals = spack.detection.by_path(["win-sdk", "wgl"])
+    if not externals:
+        raise RuntimeError(
+            "Unable to find the Windows SDK, please install via the Visual Studio installer\
+before proceeding with Spack or provide the path to a non standard install via\
+'spack external find --path'"
+        )
+    # wgl/sdk are not required for bootstrapping Spack, but
+    # are required for building anything non trivial
+    # add to user config so they can be used by subsequent Spack ops
+    spack.detection.update_configuration(externals, scope="user", buildable=False)
```

Comment:
I specifically did not want to limit this to and env or the bootstrapping scope for exactly the reasons you mention. 
The user scope was because I thought it was considered less invasive, but I'm happy to do this at the platform scope (although that modifies tracked files right?)

Reply:
platform scope refer to a set of scopes, I think perhaps you mean the site scope (i.e. that lives in `<spack-prefix>/etc/defaults`, which also includes platform-specific sub-scopes) - that is tracked yes.

My current preference is not to specify any scope (which will conditionally modify either the user or the env scope depending on where we are).

Reply:
If we were to use no scope in an env, this would force a rebootstrap (of just this part) outside of that env. Can we scope this to both the bootstrap and no scope (user)? I'd like to avoid having to repeatedly run this in every env and outside of those envs.